addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -295,13 +295,21 @@ odoo.define('website.s_website_form', function (require) {
             // force server date format usage for existing fields
             this.$target.find('.s_website_form_field:not(.s_website_form_custom)')
             .find('.s_website_form_date, .s_website_form_datetime').each(function () {
+                const $input = this.querySelector('input');
+
+                // Datetimepicker('viewDate') will return `new Date()` if the
+                // input is empty but we want to keep the empty value
+                if (!$input.val()) {
```

Comment:
This is probably making the code crashing, `.val()` is a jQuery method, it is not working on DOM elements.
Use `.value`

Reply:
Yeah look like I got trolled by the cache because it worked when I tried it 😅